cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -122,6 +133,24 @@ public static DataprocClient fromConf(DataprocConf conf) throws IOException, Gen
     String subnet = conf.getSubnet();
     Network networkInfo = getNetworkInfo(projectId, network, compute);
 
+    boolean useInternalIP;
+    
+    // if a peered network is specified then the instance is tied to a network
+    PeeringState state = peeringState(systemNetwork, systemProjectId, networkInfo);
+    if (state == PeeringState.ACTIVE) {
+      if (conf.isPreferExternalIP()) {
+        LOG.warn("Prefer external IP setting will be ignored and internal IP will used to communicate with the " +
+                   "Dataproc cluster as active peering with the system network found.");
+      }
+      useInternalIP = true;
+    } else if (state == PeeringState.INACTIVE) {
+      throw new IllegalStateException(String.format("VPC Peering to the system network %s in system project %s found " +
```

Comment:
should include the dataproc network being used.

VPC Peering from network '%s' in project '%s' to network '%s' in project '%s' is in the INACTIVE state. Please fix the peering setup to be in the ACTIVE state.

Reply:
fixed.